portfolio/src/main/webapp/js/background.js

```diff
@@ -0,0 +1,291 @@
+/**
+ * @fileoverview Creates a live visualization of balls that repel each other
+ * and bounce off of the walls.
+ */
+const HEIGHT_TO_WIDTH = 0.3;
+const EFFECT_RADIUS_TO_WIDTH = 0.1;
+const BALL_SIZE_TO_WIDTH = 0.01;
+const NUM_BALLS = 25;
+const INITIAL_BALL_VELOCITY_MAGNITUDE = 1;
+/**
+ * Runs initial set up of the background and background canvas.
+ */
+function setup() {
+  const backgroundContainer = document.getElementById("background-container");
+
+  const backgroundCanvas =
+    createCanvas(backgroundContainer.clientWidth,
+      backgroundContainer.clientWidth * HEIGHT_TO_WIDTH);
+  backgroundCanvas.parent("background-container");
+
+  frameRate(60);
+
+  initTank();
+}
+
+let ballTank;
+/**
+ * Initializes a tank for the background.
+ */
+function initTank() {
+  const ballSize = BALL_SIZE_TO_WIDTH * width;
+  ballTank = new Tank(width, height, EFFECT_RADIUS_TO_WIDTH * width, ballSize);
+
+  // Create balls and add them to the tank.
+  // Each start with random positions and velocities.
+  for (let i = 0; i < NUM_BALLS; i++) {
+    const randomPos = createVector(Math.random() * ballTank.width,
```

Comment:
I think as Joe commented we prefer randomPosition here?

Reply:
You're right. Let me go through and fix the remaining pos abbreviations.